lib/fs/basicfs_copy_range.go

```diff
@@ -0,0 +1,57 @@
+// Copyright (C) 2019 The Syncthing Authors.
+//
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this file,
+// You can obtain one at https://mozilla.org/MPL/2.0/.
+
+package fs
+
+import (
+	"syscall"
+
+	"github.com/syncthing/syncthing/lib/sync"
+)
+
+var (
```

Comment:
Everything in this file except for the last method in principle isn't basicfs specific, i.e. should live in filesystem.go or filesystem_copy_range.go. Same for the methods listing file and the standard implementation.